linmod/eval.py

```diff
@@ -1,46 +1,102 @@
 import polars as pl
 
-from linmod.utils import pl_mae
+from linmod.utils import pl_norm
 
 
-def proportions_mae_per_division_day(samples, data) -> pl.DataFrame:
+def _merge_samples_and_data(samples, data):
```

Comment:
I see this as a special case of a more general step. We will also eventually want to merge samples of predictive counts with the actual count data. So the general case is "combine data and samples for lineage observations" and the special cases are "combine frequencies" (this function) and "combine counts."

My concern is that, down the road, we'll forget which case this function is actually handling and end up with a bug that we may not even notice. I'd advocate either renaming this function slightly (`_merge_samples_and_data_phi` or some such) to disambiguate which case or generalizing (e.g. `_merge_samples_and_data(samples, data, counts=False)`, even if `counts=True` only exists to raise a `NotImplemetedError`).

Reply:
Good point. My favorite idea so far is to have submodules for the `eval` module, e.g. `eval.proportions`, `eval.counts`, etc., to have functions such as `eval.proportions.energy_score_by_division_day` and `eval.counts._merge_samples_and_data`. This way, we can keep the actual function names tidy?

Reply:
As a naming convention, I am not opposed to this. Though I am not an expert on what's "pythonic" or not.

For a function like this, having one in each wouldn't bother me either. But in general, I do want to make sure that we keep code duplication minimal. I'm not sure how many things we will actually want to do in both proportion-land and count-land. We will have to keep an eye on this.

Reply:
I don't know that we need to over-engineer super-hard on this.

I think the real solution would be to have the functions return different objects, like a PhisSample or CountsSample, which would then have a method `.merge_whatever()`.

This object could also do the validation, to make sure that it has the right columns, etc.

But that's further down the line.